e2e/testcases/permission_test.go

```diff
@@ -0,0 +1,387 @@
+		},
+		{
+			name:           "GSA - Git source missing IAM binding for Fleet WI",
+			fwiTest:        true,
+			sourceType:     v1beta1.GitSource,
+			AuthType:       configsync.AuthGCPServiceAccount,
+			sourceRepo:     csrRepo(),
+			gsaEmail:       gsaRestrictedEmail(),
+			expectedErrMsg: missingWIBindingError,
+		},
+		{
+			name:           "GSA - OCI source missing IAM binding for Fleet WI",
+			fwiTest:        true,
+			sourceType:     v1beta1.OciSource,
+			AuthType:       configsync.AuthGCPServiceAccount,
+			sourceRepo:     privateARImage(),
+			gsaEmail:       gsaRestrictedEmail(),
+			expectedErrMsg: missingWIBindingError,
+		},
+		{
+			name:           "GSA - Helm source missing IAM binding for Fleet WI",
+			fwiTest:        true,
+			sourceType:     v1beta1.HelmSource,
+			AuthType:       configsync.AuthGCPServiceAccount,
+			sourceVersion:  privateCoreDNSHelmChartVersion,
+			sourceChart:    privateCoreDNSHelmChart,
+			gsaEmail:       gsaRestrictedEmail(),
+			expectedErrMsg: missingWIBindingError,
+		},
+		{
+			name:           "GSA - Git source with IAM binding for GKE WI, but missing reader permission",
+			fwiTest:        false,
+			sourceType:     v1beta1.GitSource,
+			AuthType:       configsync.AuthGCPServiceAccount,
+			sourceRepo:     csrRepo(),
+			gsaEmail:       gsaRestrictedImpersonatedEmail(),
+			expectedErrMsg: missingCSRPermissionError,
+		},
+		{
+			name:           "GSA - OCI source with IAM binding for GKE WI, but missing reader permission",
+			fwiTest:        false,
+			sourceType:     v1beta1.OciSource,
+			AuthType:       configsync.AuthGCPServiceAccount,
+			sourceRepo:     privateARImage(),
+			gsaEmail:       gsaRestrictedImpersonatedEmail(),
+			expectedErrMsg: missingOCIARPermissionError,
+		},
+		{
+			name:           "GSA - Helm source with IAM binding for GKE WI, but missing reader permission",
+			fwiTest:        false,
+			sourceType:     v1beta1.HelmSource,
+			AuthType:       configsync.AuthGCPServiceAccount,
+			sourceVersion:  privateCoreDNSHelmChartVersion,
+			sourceChart:    privateCoreDNSHelmChart,
+			gsaEmail:       gsaRestrictedImpersonatedEmail(),
+			expectedErrMsg: missingHelmARPermissionError,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			nt := nomostest.New(t, nomostesting.WorkloadIdentity, ntopts.Unstructured, ntopts.RequireGKE(t))
+			rs := fake.RootSyncObjectV1Beta1(configsync.RootSyncName)
+			tenant := "tenant-a"
+
+			if err := workloadidentity.ValidateEnabled(nt); err != nil {
+				nt.T.Fatal(err)
+			}
+
+			mustConfigureMembership(nt, tc.fwiTest, false)
+
+			spec := &sourceSpec{
+				sourceType:    tc.sourceType,
+				sourceRepo:    tc.sourceRepo,
+				sourceChart:   tc.sourceChart,
+				sourceVersion: tc.sourceVersion,
+				gsaEmail:      tc.gsaEmail,
+			}
+			mustConfigureRootSync(nt, rs, tenant, spec)
+			nt.WaitForRootSyncSourceError(rs.Name, status.SourceErrorCode, tc.expectedErrMsg)
```

Comment:
All these auth errors makes me wonder if we shouldnt add a new error code. I would normally expect the problem with a source error to be in the source, but these are all auth problems fetching the source.